### ISSUE TITLE
expression2List() not taking alphanumeric values

### ISSUE THREAD

**Issue by jdime** [ncborcherding/scRepertoire#23]
Hello,

Thanks for the package it's being very useful for my analyzes. I'm using **scRepertoire v1.2.1**. All functions work as expected following the vignette. But I have one question regarding `expression2List()`. It works when I use metadata with numeric values (e.g. 'seurat_clusters') but it doesn't work when using alphanumeric values (see example below). The list resulting from `expression2List()` has the correct `names()` for both 'seurat_clusters' and 'sample_type' metadata categories, but there are no values in the list when I use 'sample_type', even though I can see values in the Seurat object metadata.  Note both metadata properties are passed as 'factor'.

```
>   seurat.object.each_clone_call@meta.data[c(1:3,4013:4016),c("seurat_clusters","sample_type")]
                               seurat_clusters sample_type
SMTR02_pretx_CGTAGCGCAACGCACC               20       pretx
SMTR02_pretx_AGGGATGAGAGGTTGC               20       pretx
SMTR02_pretx_TCGGGACGTCGATTGT               20       pretx
SMTR07_t2_Rad_TCCACACTCCATGAAC              13         Rad
SMTR07_t2_Rad_TGAGGGACATGAGCGA              13         Rad
SMTR07_t2_Rad_TAGACCACAAGGTTCT              13         Rad
SMTR07_t2_Rad_CCGGTAGCAGGTTTCA              20         Rad

>   class(seurat.object.each_clone_call@meta.data$seurat_clusters)
[1] "factor"
>   combined.seurat <- expression2List(seurat.object.each_clone_call, group = "seurat_clusters")
>   names(combined.seurat)
[1] "0"  "5"  "11" "13" "15" "20"
>   head(combined.seurat$`0`$seurat_clusters)
[1] 0 0 0 0 0 0
Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23

>   class(seurat.object.each_clone_call@meta.data$sample_type)
[1] "factor"
>   combined.seurat <- expression2List(seurat.object.each_clone_call, group = "sample_type")
>   names(combined.seurat)
[1] "NonRad" "pretx"  "Rad"   
>   head(combined.seurat$Rad$sample_type)
factor(0)
Levels: NonRad pretx Rad
```

Is this behaviour expected? Can `expression2List()` be extended to take alphanumeric values?

Thanks.

**Comment by ncborcherding** [comment]
Hey jdime - 

Thanks for giving the version and clear problem. The issue here I think is a bug in the function - specifcally the group parameter was incorrectly called in the function itself. Due to the submission requirements for bioconductor, the scRepertoire version had to be reduced  -  v0.99.8 has the change already incorporated.

If you don't want to update the current version, should be able to just assign the following into your global environment.

```
expression2List.mod <- function(sc, group) {
    if (!inherits(x=sc, what ="Seurat") & 
        !inherits(x=sc, what ="SummarizedExperiment")) {
            stop("Use a seurat or SCE object to convert into a list")
    }
    meta <- grabMeta(sc)
    unique <- str_sort(as.character(unique(meta[,group])), numeric = TRUE)
    df <- NULL
    for (i in seq_along(unique)) {
        subset <- subset(meta, meta[,group] == unique[i])
        df[[i]] <- subset
    }
    names(df) <- unique
    return(df)
}
```

Please let me know if that works or if you had any suggestions for the package as a whole.

Thanks

**Comment by jdime** [comment]
Hello @ncborcherding,

Thanks for your prompt response.

For the record: I added `expression2List.mod()` to v1.2.1 and it asked me for `grabMeta()`, which I added from [here](https://github.com/ncborcherding/scRepertoire/blob/09eed51746591e3d16e52f27b4d8b648dadec783/R/utils.R). Then it asked me for library(stringr), which I just needed to load. After that, `expression2List.mod()` worked fine.

I'll keep using scRepertoire and will send you my suggestions in a separate thread.

Thanks.
